sdks/java/io/kinesis/src/main/java/org/apache/beam/sdk/io/kinesis/ShardReadersPool.java

```diff
@@ -44,13 +48,14 @@
 
   private static final Logger LOG = LoggerFactory.getLogger(ShardReadersPool.class);
   private static final int DEFAULT_CAPACITY_PER_SHARD = 10_000;
-  private ExecutorService executorService;
-  private BlockingQueue<KinesisRecord> recordsQueue;
-  private Map<String, ShardRecordsIterator> shardIteratorsMap;
-  private SimplifiedKinesisClient kinesis;
-  private KinesisReaderCheckpoint initialCheckpoint;
+  private final ExecutorService executorService;
+  private volatile BlockingQueue<KinesisRecord> recordsQueue;
+  private volatile ImmutableMap<String, ShardRecordsIterator> shardIteratorsMap;
+  private final ConcurrentMap<String, AtomicInteger> counters;
```

Comment:
Can we please use more descriptive name than counters?
Is it numberOfRecordsInAQueueByShard or something along that lines?